central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -198,7 +207,50 @@ func (m *managerImpl) runReports() {
 				continue
 			}
 			logging.Infof("Executing report %q at %v", req.scanConfig.GetId(), time.Now().Format(time.RFC822))
-			go m.generateReport(req)
+			var w watcher.ScanConfigWatcher
```

Comment:
Comment about the `case <-m.stopper.Flow().StopRequested():` and the semaphore - we use `context.Background()` for the semaphore and do not bind it to the stopper signal - is this intended? 

Note that the `generateReport` does not accept a context - or it does through req, but that seems to be also not connected to stopper signal nor the context used for the semaphore. Do we want the report generation to keep going even after calling the stopper?

Reply:
I see that the `req` has a context, but it is the one from grpc call with cancel stripped away. Most probably this suggests that we want to keep the report generation going under all circumstances. Please confirm/deny

Reply:
Yeah there's room for improvement here. I'm not sure if we should leave it for a follow-up or not since this was like that before.